File: src/com/uipkg/simple/SimpleFrame.java
package com.uipkg.simple;

import javax.swing.JFrame;

import com.uipkg.exceptions.NonApplicableException;

public interface SimpleFrame {
  public void run();
  public void runWithTask(Runnable s) throws NonApplicableException;
  public void killTask() throws NonApplicableException;
  public JFrame getInstanceFrame();
}

File: src/com/uipkg/simple/SimpleFrame.kt
package com.uipkg.simple

import com.uipkg.exceptions.NonApplicableException
import kotlin.Throws
import java.lang.Runnable
import javax.swing.JFrame

interface SimpleFrame {
    fun run()

    @Throws(NonApplicableException::class)
    fun runWithTask(s: Runnable?)

    @Throws(NonApplicableException::class)
    fun killTask()
    val instanceFrame: JFrame?
}